ports/esp32/mphalport.c

```diff
@@ -154,3 +156,12 @@ int *__errno() {
     return &mp_stream_errno;
 }
 */
+
+/* Wake up the main task if it is sleeping */
+void mp_hal_wake_main_task_from_isr(void) {
+    BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+    vTaskNotifyGiveFromISR(mp_main_task_handle, &xHigherPriorityTaskWoken);
+    if (xHigherPriorityTaskWoken == pdTRUE) {
+        portYIELD_FROM_ISR();
```

Comment:
Do you know if this is needed?  I'd assume that the main uPy task is lower priority than all ISRs and that it would be better to just let this ISR (eg timer, pin) finish as fast as possible by not calling this yield.  Then the main uPy task will be woken when the ISR returns (I guess, because `vTaskNotifyGiveFromISR` should have put the main uPy task back on the runable queue).

Reply:
The documentation for `vTaskNotifyGiveFromISR()` says that you need to contitionally call `portYIELD_FROM_ISR()` in case a higher priority task has been woken. I'm guessing that in our current usage this never happens (since there are no higher priority tasks than IRQ handlers) but I put it in there because the documentation said I should and we might need it in the future.

Reply:
Ok, let's keep it.